R/format_PFN.R

```diff
@@ -0,0 +1,555 @@
+  Brood_data <- Brood_data %>%
+    dplyr::select(-ChickAge)
+
+  # EXPORT DATA
+
+  time <- difftime(Sys.time(), start_time, units = "sec")
+
+  message(paste0("All tables generated in ", round(time, 2), " seconds"))
+
+  if(output_type == "csv"){
+
+    message("Saving .csv files...")
+
+    utils::write.csv(x = Brood_data, file = paste0(path, "\\Brood_data_PFN.csv"), row.names = F)
+
+    utils::write.csv(x = Individual_data, file = paste0(path, "\\Individual_data_PFN.csv"), row.names = F)
+
+    utils::write.csv(x = Capture_data %>% select(-Sex, -BroodID), file = paste0(path, "\\Capture_data_PFN.csv"), row.names = F)
+
+    utils::write.csv(x = Location_data, file = paste0(path, "\\Location_data_PFN.csv"), row.names = F)
+
+    invisible(NULL)
+
+  }
+
+  if(output_type == "R"){
+
+    message("Returning R objects...")
+
+    return(list(Brood_data = Brood_data,
+                Capture_data = Capture_data,
+                Individual_data = Individual_data,
+                Location_data = Location_data))
+
+  }
+
+}
+
+
+
+#' Create brood data table for EastDartmoor.
+#'
+#' @param db Location of primary data from EastDartmoor.
+#'
+#' @return A data frame with Brood data
+
+
+create_brood_PFN <- function(db){
+
+  ## Pre) Determine a vector of "bad" (nonconclusive) IDs
+  #This will be used downstream in point 7)
+  badIDs <- c("RUNT", "ringed", "ringed left", "ringed right", "unringed", "Unringed")
+
+  ## 0) Load data
+  #We read everything in as text and convert it afterwards
+  #Even though some columns (e.g. date) work well, they may be broken with newer data.
+  #Using text and converting manually should be more robust to data changes
+
+    Brood_data <- utils::read.csv(file = paste0(db, "/PFN_PrimaryData_EDartmoor.csv"), na.strings = c("", "?"),
+                                    colClasses = "character") %>%
+
+    ## 1) Rename columns that are equivalent (content and format) to columns in the standard format
+    dplyr::rename(Plot = Popn,
+                  LocationID = Box) %>%
+
+    ## 2) Add population ID and reformat colums with equivalent content but different format
+    dplyr::mutate(PopID = "PFN",
+                  BreedingSeason = as.integer(Year),
+                  Species = dplyr::case_when(Species == "BLUTI" ~ "CYACAE",
+                                            Species == "PIEFL" ~ "FICHYP",
+                                            Species == "GRETI" ~ "PARMAJ",
+                                            Species == "REDST" ~ "PHOPHO",
+                                            Species == "MARTI" ~ "POEPAL",
+                                            Species == "NUTHA" ~ "SITEUR",
+                                            Species == "COATI" ~ "PERATE",
+                                            Species == "WREN" ~ NA_character_, # Missing, 1 observation only
+                                            Species == "TREEC" ~ NA_character_), # Missing, 1 observation only
```

Comment:
Good to know! I think in this case it may still be good to keep it in, in case someone wonders what happened to the "WREN" and "TREEC" entries. But if you'd rather remove it, we can do that too.